Meme/Meme/MemeParser.swift

```diff
@@ -0,0 +1,30 @@
+//
+//  MemeParser.swift
+//  Meme
+//
+//  Created by Mesfin Bekele Mekonnen on 7/29/19.
+//  Copyright © 2019 Mesfin Bekele Mekonnen. All rights reserved.
+//
+
+import Foundation
+
+struct MemeParser {
+
+    func parse(imageData: Data, memeName: String) -> Meme? {
+            do {
+                if let jsonResponse = try JSONSerialization.jsonObject(with: imageData, options: []) as? [String:Any] {
+                    if let imageURLDict = jsonResponse["direct"] as? [String:String] {
+                        if let visibleImageString = imageURLDict["visible"] {
+                            if let imageURL  = URL(string: visibleImageString) {
+                                return Meme(name: memeName, imageURL: imageURL)
+                            }
+                        }
+                    }
+                }
```

Comment:
So one of the nice things about swift is you can chain `if let` and guard clauses. That means you can rewrite this without nesting:
```swift
guard let jsonResponse = try? JSONSerialization.jsonObject(with: imageData, options:[]) as? [String: Any],
    let imageURLDict = jsonResponse["direct"] as? [String: String],
    let visibileImageString = imageURLDict["visible"],
    let imageURL = URL(string: visibleImageString)
    else { print("Error decoding values"; return nil }

return Meme(name: memeName, imageURL: imageURL)
```

Reply:
using `try?` ops for returning an optional JSON/nil instead of JSON or throwing an Error. That makes it easier to add the try? to a guard/if let. The Error can be useful sometimes however, so just have to weigh the options.

Reply:
Nice, I didn't know that. Looks way cleaner that mine. I'll update it, thanks for the suggestion! As for the error, I think let's start with just returning nil for now and we can update it if we think the error is useful. It's only catching the error and printing anyway in this PR.